source/adapters/cuda/enqueue.cpp

```diff
@@ -426,26 +457,33 @@ UR_APIEXPORT ur_result_t UR_APICALL urEnqueueKernelLaunch(
   size_t BlocksPerGrid[3] = {1u, 1u, 1u};
 
   uint32_t LocalSize = hKernel->getLocalSize();
-  ur_result_t Result = UR_RESULT_SUCCESS;
   CUfunction CuFunc = hKernel->get();
 
-  Result = setKernelParams(hQueue->getContext(), hQueue->Device, workDim,
-                           pGlobalWorkOffset, pGlobalWorkSize, pLocalWorkSize,
-                           hKernel, CuFunc, ThreadsPerBlock, BlocksPerGrid);
-  if (Result != UR_RESULT_SUCCESS) {
-    return Result;
-  }
+  UR_CHECK_ERROR(setKernelParams(hQueue->getContext(), hQueue->Device, workDim,
+                                 pGlobalWorkOffset, pGlobalWorkSize,
+                                 pLocalWorkSize, hKernel, CuFunc,
+                                 ThreadsPerBlock, BlocksPerGrid));
 
   try {
     std::unique_ptr<ur_event_handle_t_> RetImplEvent{nullptr};
 
+    ScopedContext Active(hQueue->getDevice());
     uint32_t StreamToken;
     ur_stream_guard_ Guard;
     CUstream CuStream = hQueue->getNextComputeStream(
         numEventsInWaitList, phEventWaitList, Guard, &StreamToken);
 
-    Result = enqueueEventsWait(hQueue, CuStream, numEventsInWaitList,
-                               phEventWaitList);
+    if (DepEvents.size()) {
```

Comment:
!DepEvents.empty() ?

Reply:
I think this is equivalent and `.size()` is more readable than `!....empty()`

Reply:
performance wise using empty() should be preferred,  I don't have hard feeling for that though.

Reply:
my impression is that empty() should be implemented as `return begin() == end()`. but size() has to count.  I have no hard feeling about that and depending on size one may be more performant.